tests/test_io_utils.py

```diff
@@ -0,0 +1,183 @@
+import os
+import tempfile
```

Comment:
Testing write operations can be tricky. Ideally, we'd like to avoid writing to the file system, even using tempfiles. It can be inconsistent when running unit tests in docker containers. Usually people use the mock library to mock the write operation without actually writing anything.

Reply:
I see, will look into that